File: src/console/command/parameter/def/BoolParameter.php
<?php

namespace pocketcloud\cloud\console\command\parameter\def;

use pocketcloud\cloud\console\command\parameter\CommandParameter;

readonly class BoolParameter extends CommandParameter {

    public function __construct(string $name, bool $optional, ?string $customErrorMessage = null) {
        parent::__construct($name, $optional, $customErrorMessage);
    }

    public function parseValue(string $input): bool {
        if (strtolower($input) == "true" || strtolower($input) == "yes") return true;
        return false;
    }

    public function onTabCompleteMatch(string $currentArg): array {
        return ["true", "false"];
    }

    public function getType(): string {
        return "boolean";
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/parameter/def/BoolParameter.java
package de.pocketcloud.cloud.console.command.parameter.def;

import de.pocketcloud.cloud.console.command.parameter.BaseCommandParameter;

import java.util.List;

public class BoolParameter extends BaseCommandParameter {

    public BoolParameter(String name, boolean optional, String customErrorMessage) {
        super(name, optional, customErrorMessage);
    }

    public BoolParameter(String name, boolean optional) {
        super(name, optional);
    }

    @Override
    public Boolean parseValue(String input) {
        return input.equalsIgnoreCase("true") || input.equalsIgnoreCase("yes");
    }

    @Override
    public List<String> onTabCompleteMatch(String currentArg) {
        return List.of("true", "false");
    }

    @Override
    public String getType() {
        return "boolean";
    }
}